src/lib/y2network/wicked/connection_config_readers/base.rb

```diff
@@ -158,14 +159,37 @@ def all_ips
         # It takes an IP address and, optionally, a prefix or a netmask.
         #
         # @param ip      [Y2Network::IPAddress] IP address
-        # @param prefix  [Integer,nil] Address prefix
-        # @param netmask [String,nil] Netmask
-        def build_ip(ip, prefix, netmask)
+        # @param id      [String] Hash key for the IP Address
+        def build_ip(ip, id)
           ipaddr = ip.clone
           return ipaddr if ip.prefix?
 
-          ipaddr.netmask = netmask if netmask
-          ipaddr.prefix = prefix if prefix
+          begin
+            netmask = file.netmasks[id]
```

Comment:
np: what about moving those begin/rescue blocks to smaller methods? This method has more than 20 lines and most of the code is about registering an issue.